### DISCUSSION TITLE
Interop with Async/Tasks

### DISCUSSION THREAD

**Discussion by kthompson** [LGouellec/streamiz#135] [General]
Is there a general or recommended pattern for mixing Async/Tasks with Streams? For example in one of my streams I would like to add as the last step a `.Foreach` but in that I need to send an async http request. Are there any suggestions/docs on how to do this?

**Comment by LGouellec** [comment]
Hi,

Calling external system inside your topology Kafka streams is not a good practice. [see](https://stackoverflow.com/questions/70723169/can-kafka-streams-interact-with-external-systems).

Best pattern is publish your sink record into a topic and use a HttpSjnkConnector and forward to your web service 

Kr,

**Reply by LGouellec** [reply]
Stay tuned #147
